staging/src/k8s.io/apiserver/pkg/util/flowcontrol/gen_test.go

```diff
@@ -570,23 +574,29 @@ func genResourceRule(rng *rand.Rand, pfx string, mayMatchClusterScope, matchAllR
 		rr.ClusterScope = true
 		rr.Namespaces = []string{flowcontrol.NamespaceEvery}
 	}
+	if matchAllResources || starMask&16 == 16 && rng.Float32() < 0.1 {
+		rr.ResourceNames = []string{flowcontrol.ResourceNameEvery}
+	}
 	return rr, matchingRIs, skippingRIs
 }
 
-func genRRIs(rng *rand.Rand, m int, verbs, apiGroups, resources, namespaces []string) []*request.RequestInfo {
+func genRRIs(rng *rand.Rand, m int, verbs, apiGroups, resources, namespaces, names []string) []*request.RequestInfo {
 	nv := len(verbs)
 	ng := len(apiGroups)
 	nr := len(resources)
-	nn := len(namespaces)
-	coords := chooseInts(rng, nv*ng*nr*nn, m)
+	nns := len(namespaces)
+	nn := len(names)
+	coords := chooseInts(rng, nv*ng*nr*nns*nn, m)
 	ans := make([]*request.RequestInfo, 0, m)
 	for _, coord := range coords {
 		ans = append(ans, &request.RequestInfo{
 			IsResourceRequest: true,
 			Verb:              verbs[coord%nv],
 			APIGroup:          apiGroups[coord/nv%ng],
 			Resource:          resources[coord/nv/ng%nr],
-			Namespace:         namespaces[coord/nv/ng/nr]})
+			Namespace:         namespaces[coord/nv/ng/nr%nns],
+			Name:              names[coord/nv/ng/nr/nns%nn],
```

Comment:
The remainder operation is not needed here, just as it was not needed at the end of the original series.